seed_data/lib.py

```diff
@@ -250,25 +253,36 @@ def set_data(self, obj, course_run, grade=DEFAULT_GRADE, **kwargs):
 }
 
 
-def clear_edx_data_for_course_run(user, course_run, models=None):
-    """Clears all of a User's cached edX data for a CourseRun, or just the data for specific cached models"""
-    if models:
-        handlers = [CACHED_HANDLERS[model] for model in models]
-    else:
-        handlers = CACHED_HANDLERS.values()
-    for cached_model_handler in handlers:
-        cached_model_handler(user).clear_all(course_run=course_run)
-
-
-def clear_edx_data_for_course(user, course, models=None):
-    """Clears all of a User's cached edX data for a Course, or just the data for specific cached models"""
+def ensure_cached_edx_data_for_user(user):
+    """
+    Ensures that all possible cached edX records have been initialized for a User
+    """
+    all_course_runs = CourseRun.objects.filter(course__program__live=True).exclude(
+        Q(edx_course_key__isnull=True) | Q(edx_course_key__exact='')
+    ).all()
+    all_course_run_ids = set([course_run.id for course_run in all_course_runs])
+    results = []
+    for cached_model in [CachedEnrollment, CachedCertificate, CachedCurrentGrade]:
+        all_cached_enrollments = cached_model.objects.filter(user=user).values_list("course_run__id", flat=True)
+        uninit_course_run_ids = set(all_course_run_ids).difference(set(all_cached_enrollments))
```

Comment:
`all_course_run_ids` is already a set. `all_cached_enrollments` isn't, but you could easily make it one when it is defined. Then you can just do operations on these sets, without calling `set()` on them again.